content/How to Make My First Gem - Timothy Peraza.md

```diff
@@ -0,0 +1,128 @@
+
+<img src="images/rsgem_structure.png"
+     alt="Blog Gem"
+     width="500" height="250"
+     align="middle"/>
+
+It's important to mention that the gem rsgem is based on the command 'bundle gem', said command is the one that generates the directories and the files that we see. Now, we will mention the different components which are generated for the gem that we want to create.
+
+## Bin folder
+
+<img src="images/bin_folder.png"
+     alt="Bin folder"
+     width="500" height="250"
+     align="middle"/>
+
+In the Bin folder, we can find the console and setup files. The important thing to highlight here is the utility of console file where we can see the next code:
+
+
+```ruby
+#!/usr/bin/env ruby
+
+require 'bundler/setup'
+require 'blog_gem'
+
+# You can add fixtures and/or initialization code here to make experimenting
+# with your gem easier. You can also use a different console, if you like.
+
+# (If you use this, don't forget to add pry to your Gemfile!)
+# require "pry"
+# Pry.start
+
+require 'irb'
+IRB.start(__FILE__)
+```
+
+If we have to do test from the terminal with our gem, we only need the command bundle console 'require_name' to be able to access to the interactive ruby shell with each of our methods and dependecies of the gem. In our example of gem, we have just to do bundle console blog_gem.
+
+## Lib folder
+
+<img src="images/lib_folder.png"
+     alt="Lib folder"
+     width="500" height="250"
+     align="middle"/>
+
+In the Lib folder is where the magic will be done, here we will put our classes, our code, our methods that will be offered to those developers who want to use the gem that we make. If we look closely, we will see a folder with the name of our gem and if we look inside the folder, we will find a file with the name version. This version file is used to indicate what is the version of our gem and it is the file that is used to complete the version attribute of our gemspec file which we will see later.
+
+On the other hand we have the file name_gem.rb (in our example 'blog_gem.rb'), which is the main file of the gem. This file contains code similar to the following:
+
+```ruby
+require 'blog_gem/version'
+
+module BlogGem
+  class Error < StandardError; end
+  # Your code goes here...
+end
+
+```
+
+As we can see, it is a class where, in principle, you could put code directly. That decision is up to the developer's wish, but generally no code is added to this file, instead of it, 'requires' are done to the files which contain the classes which offer the methods of our gem. We can also see this class does generic error handling which follows the ruby guidelines.
```

Comment:
```suggestion
As we can see here, it is a class wherein principle, you could put the code directly in. That decision is up to the developer, but generally, there won't be code added to this file. Instead of this, 'requires' will be added to the files, which contain the classes which offer the methods of our gem. We can also see this class covers generic error handling, which follows the ruby guidelines.
```